### ISSUE TITLE
Pre-preregistration questions

### ISSUE THREAD

**Issue by Jaeoc** [chartgerink/2016marginal#22]
@chartgerink 

Hi! Some questions/concerns that I would appreciate your opinion on:
-----------------------------------------------------------------
- Marcel pointed out that we seem to have a higher percentage of marginally significant results than Pritschet et al, which is a bit unfortunate. Unless there's some mistake in the code that I've missed, I assume  this is because we get some overlap with our 200 character search, although we would have to wait for the full data of the years that we can compare directly with pritschet et al to be sure. Any advice on how to deal with it if that turns out to be the case?

- The 'core of psychology' stuff: I eliminated it as a category, but it seemed a waste of data not to use the journals when doing the estimations for 'All APA journals' since they're logically still part of that category. Do you think I should exclude those four journals and their content completely?

- I was thinking of deleting the part of the readme.md file on github that begins "In order to create a more managable file for analyses" since that doesn't seem relevant anymore. Objection?

- Is it standard to upload a text-file (or maybe put it in the readme) describing the different variables in the datasets, as Pritschet et al did?
-----------------------------------------------------------------

Hope the wall of text is not too intimidating, as you've said, I can be a bit wordy sometimes ;)

**Comment by chartgerink** [comment]
1. I agree with your reasoning about the 200 characters. We have to realize it's quite a lot of surrounding text, 400 characters:
> xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx p=.08 xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

As such, our estimate will be quite liberal, resulting in higher numbers of found results reported as marginal.
2. Well it is rather confusing, so either make it clearer why you include them despite excluding the "Core of psychology" one, or be consistent I'd say. 
3. Yes, has become redundant indeed. Good catch! (leaving it would be confusing)
4. A codebook is always necessary, yes. Check out [this](https://github.com/chartgerink/2015ori-3/tree/master/data) folder where I include `funnels.csv` (data file) and its codebook `funnels.csv.md`. Always important to describe the variables and what their properties are. Some people create word files, but text files are easier :100:

Don't worry, you structured in bullet points, that helps :smile: You can close the issue if this answers your questions, but feel free to discuss further if you want to have something clarified.

**Comment by Jaeoc** [comment]
1. Hmm.. Can we use regular expressions to find out average sentence length in characters, by for example counting the number of characters between each dot (".") that is immediately followed by space and a capital letter? (e.g. ". H") If so, we could then also find out how far away from the end/beginning of a sentence a p-value is on average. Perhaps something for some additional analyses if possible? (I was trying to find a previous study on average sentence lengths in scientific papers but couldn't find anything)

2,3,4: ok, thanks!

**Comment by chartgerink** [comment]
I understand what your goal for that would be, but then I'd recommend just checking how far away the `(margin*|approach*)` is from the p-value. It would pretty much get at the same thing. You could do that, yes, as a sensitivity analysis. You'd need to take a look at the `regexpr` function I think. See some example code below that searches for the `y` in the string of `x`s.

```bash
> x <- 'xxxxxxxyxxxxxx p = .05 xxxxxxxxxxxxxxxxxxxx'
> grep(
pattern=      ignore.case=  value=        useBytes=     
x=            perl=         fixed=        invert=       
> grep(pattern = 'y', x)
[1] 1
> regexpr(pattern = 'y', x)
[1] 8
attr(,"match.length")
[1] 1
attr(,"useBytes")
[1] TRUE
```

As you see, the return says the position of the match 'y' is the 8th position. It only finds the first one though:
```bash
> x <- 'xxxxxxxyxxxxxx p = .05 xxxxxxxxxxxxxxxxxyxxx'
> regexpr(pattern = 'y', x)
[1] 8
attr(,"match.length")
[1] 1
attr(,"useBytes")
[1] TRUE
```

You could use the `stringr::str_locate_all` if you want to find all hits in a string
```bash
> stringr::str_locate_all(pattern = 'y', x)
[[1]]
     start end
[1,]     8   8
[2,]    41  41
```

Does that help?

**Comment by Jaeoc** [comment]
Clever. Hmm, should lead to some improvement. Say we base our new character limit on the average of how close the closest "marginal/approach" is to a p-value, we should be able to decrease overlap a bit, since the "marginal/approach" for the overlaps must be further away from their p-values on average. It will be a bit biased by the overlapping cases we have, but perhaps that's counteracted by just using the average distance from the p-values rather than the average + a bit as I was thinking originally. 

However, curious why you think this is preferable to examining the sentence lengths and distances of p-values from the beginning/end? Seems like that would give us more exact/useful values and about the same amount of work (unless I'm being naive here)

**Comment by Jaeoc** [comment]
Additional question: looking through your feedback, I don't really understand what you mean by this comment:

>Use the SessionInfo() function to log all version numbering and we can refer to it. :)

Do you mean put everything that SessionInfo gives me in a text-file?

**Comment by chartgerink** [comment]
- P-values location: I think more steps will introduce more errors, which is why simpler is easier. Also, what position in those 200 is more clear in my opinion. Just a suggestion, you're free to add as you like for the FYP and we'll see what we'll do. Your paper in the end, don't be afraid to do stuff we initially don't necessarily agree on. Dissidence is key to progress ;-)
- SessionInfo: Look at the end of [this](https://github.com/chartgerink/2015poldermans/blob/master/submission/manuscript.pdf) file, that's what I mean. The code for that:
```{r, echo = FALSE}
sessionInfo()
```
in an R code block
- I don't use any python here. The specific scripts are in the `functions/` folder and the command line calls for those are in the README. I don't think you have shell commandline experience, right? That makes things difficult to understand and you don't have to worry about those for the FYP (although I can of course explain if you want to understand :) I'll have time for that in september)

**Comment by Jaeoc** [comment]
- I'll think about it and see what I try :)

- Do you mean put the SessionInfo at the end of the paper in the same way? Or put it in an r-script and refer to that?

- Nope, no command line experience. Well for now, I've just referred to the readme at certain points in the text..

- I can't seem to do the preregistration (or else I misunderstand where to do it):
>Only project administrators can initiate registrations.

Do it for me, or give me privileges to do it?

**Comment by chartgerink** [comment]
Okay great. See the ref I mentioned, it puts it into the Rmarkdown file in a code chunk.
I am giving you admin rights in the next minute, please adjust the title of the project before regitering :100:

**Comment by Jaeoc** [comment]
I have a coding question, it baffles me, but perhaps it is obvious to you:

> unique(dat$marginal)
[1] FALSE  TRUE
length(unique(dat$doi))
[1] 18400
length(unique(dat$doi[dat$marginal == 0]))
[1] 13654
length(unique(dat$doi[dat$marginal == 1]))
[1] 8675
13654 + 8675
[1] 22329

How can these values add up to more than the total?

**Comment by chartgerink** [comment]
I can't really debug without the object itself. It can be due to various factors.
1. are `dat$marginal` and `dat$doi` exactly the same length?
2. is length the thing you want from unique()?
3. does `dat$doi[dat$marginal == 0]` return what you think it should return?
4. print the objects step by step and see if you anything weird.

**Comment by Jaeoc** [comment]
Resolved.
